app/crate/actors.py

```
@@ -184,6 +188,58 @@ def _release_db_heavy_lock(task_id: str):
+        if not r:
+            return True
+        deadline = time.time() + timeout
+        while time.time() < deadline:
+            current = r.scard(_DOWNLOAD_SEM_KEY)
+            if current < _DOWNLOAD_SEM_MAX:
+                r.sadd(_DOWNLOAD_SEM_KEY, task_id)
+                r.expire(_DOWNLOAD_SEM_KEY, _DOWNLOAD_SEM_TTL)
+                return True
+            time.sleep(10)
+        return False
+    except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)